homeassistant/components/history.py

```diff
@@ -96,23 +118,23 @@ def get_states(utc_point_in_time, entity_ids=None, run=None):
         func.max(states.state_id).label('max_state_id')
     ).filter(
         (states.created >= run.start) &
-        (states.created < utc_point_in_time)
-    )
-
-    if entity_ids is not None:
-        most_recent_state_ids = most_recent_state_ids.filter(
-            states.entity_id.in_(entity_ids))
+        (states.created < utc_point_in_time) &
+        (~states.domain.in_(IGNORE_DOMAINS)))
+    most_recent_state_ids = _set_filters_in_query(
```

Comment:
The get `states_to_json()`method uses the `get_states()` to get the state at start time. If we do not filter here all the filtered entities will show up again in the history, but the only show the start value (a lot of flat lines and bars with one value for the whole day).